server/src/main/java/org/apache/cassandra/sidecar/cluster/instance/InstanceMetadata.java

```diff
@@ -66,6 +72,26 @@ public interface InstanceMetadata
      */
     @NotNull CassandraAdapterDelegate delegate() throws CassandraUnavailableException;
 
+    /**
+     * @return commitlog directory of the cassandra instance
+     */
+    @Nullable String commitlogDir();
```

Comment:
I feel these should be not-null, but we can default to the Cassandra's defaults:
```
    /**
     * Returns the commit log directory of the Cassandra instance. If not set, the default directory is
     * {@link #cassandraHomeDir()}/data/commitlog
     *
     * @return commitlog directory of the cassandra instance
     */
    @NotNull
    default String commitlogDir()
    {
        return String.format("%s/data/commitlog", cassandraHomeDir());
    }
```

Reply:
It is needed only for CEP-40, so I thought of not making it mandatory. Now marking it @NotNull since we are making cassandra_home_dir mandatory. `String.format` hurts performance, so replaced with normal string concatenation.